src/libs/Navigation/Navigation.js

```diff
@@ -74,9 +81,18 @@ function closeDrawer() {
  */
 function getDefaultDrawerState(isSmallScreenWidth) {
     if (didTapNotificationBeforeReady) {
-        return 'closed';
+        return CONST.DRAWER_STATUS.CLOSED;
     }
-    return isSmallScreenWidth ? 'open' : 'closed';
+
+    if (isSmallScreenWidth) {
+        const path = getPathFromState(navigationRef.current.getState(), linkingConfig.config);
+
+        // If the initial route path is HOME SCREEN,
+        // return open for default status drawer instead of using value from Onyx
```

Comment:
**suggestion:** This comment should explain why the "home" route will always return a status of "open"

But that kind of begs the question of why `defaultDrawerStatus` is not already the correct value?

Reply:
> suggestion: This comment should explain why the "home" route will always return a status of "open"

Yeah, I will update the comment to make it more make sense.

> But that kind of begs the question of why defaultDrawerStatus is not already the correct value?

When the user opens from URL `https://staging.new.expensify.com` I make it will open the LHN even if the latest drawer status is `closed`.

Reply:
>When the user opens from URL https://staging.new.expensify.com I make it will open the LHN even if the latest drawer status is closed.

That makes sense. But wouldn't the opposite be true as well? If we are directly navigating to a "report route" then the drawer should always be closed.

I'm not sure I understand the case where we need to track and save the "last drawer status"?

Seems like if we did the following there would be no issue:

When the url is `/r/<reportID>` drawer should init closed
When the url is `/` drawer should init open

?

Reply:
I have suggestion in [here](https://github.com/Expensify/App/pull/9076#discussion_r877689623).